kite-tools-parent/kite-tools/src/main/java/org/kitesdk/cli/commands/TarImportCommand.java

```diff
@@ -0,0 +1,167 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kitesdk.cli.commands;
+
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.generic.GenericRecordBuilder;
+import org.apache.commons.compress.compressors.bzip2.BZip2CompressorInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.IOUtils;
+import org.kamranzafar.jtar.TarEntry;
+import org.kamranzafar.jtar.TarInputStream;
+import org.kitesdk.data.Dataset;
+import org.kitesdk.data.DatasetDescriptor;
+import org.kitesdk.data.DatasetWriter;
+import org.kitesdk.data.spi.DatasetRepository;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.util.List;
+
+@Parameters(commandDescription="Import files in tarball into a Dataset")
+public class TarImportCommand extends BaseDatasetCommand {
+
+  private static final List<String> SUPPORTED_TAR_COMPRESSION_TYPES = Lists.newArrayList("none", "gzip", "bzip2");
+
+  public TarImportCommand(Logger console) {
+    super(console);
+  }
+
+  @Parameter(description="<tar path> <dataset name>")
+  List<String> targets;
+
+  @Parameter(names="--compression", description="Compression type (none, gzip, bzip2)")
+  String compressionType = "none";
+
+  @Override
+  public int run() throws IOException {
+    Preconditions.checkArgument(targets != null && targets.size() == 2,
+        "Tar path and target dataset name are required.");
+
+    Path source = qualifiedPath(targets.get(0));
+    FileSystem sourceFS = source.getFileSystem(getConf());
+    Preconditions.checkArgument(sourceFS.exists(source),
+        "Tar path does not exist: " + source);
+
+    Preconditions.checkArgument(SUPPORTED_TAR_COMPRESSION_TYPES.contains(compressionType),
+        "Compression type " + compressionType + " is not supported");
+
+    String datasetName = targets.get(1);
+
+    int success = 0;
+
+    DatasetRepository repo = getDatasetRepository();
+
+    Dataset<GenericRecord> tarImport = null;
+    if (!repo.exists(namespace, datasetName)) {
+      console.info("Creating dataset {} in repo {}", datasetName, repo.getUri());
+      tarImport = repo.create(namespace, datasetName,
```

Comment:
Other import commands require that the dataset already exists. In this case, I think it makes sense to automatically create it because the schema is static. I'm not sure how well that fits with the rest of the tool though. Maybe this would make sense as a its own Kite-based CLI tool?